beamer/chains.py

```diff
@@ -0,0 +1,106 @@
+from dataclasses import dataclass
+from beamer.typing import ChainId
+
+
+@dataclass(frozen=True)
+class ChainDescriptor:
+    id: ChainId
+    ecosystem_name: str
+    name: str
+    type2: bool = False
+    bedrock: bool = False
+    testnet: bool = False
+    local: bool = False
+
+    @property
+    def supports_type_2_transactions(self) -> bool:
+        return self.type2 and not self.local
+
+    @property
+    def is_testnet(self) -> bool:
+        return self.testnet
```

Comment:
I think we don't need properties like these since we already have public fields.